src/ErrorHandling/ErrorModal.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useEffect, useState } from "react";
+import { makeStyles } from "tss-react/mui";
+import { Modal } from "../ui/React/Modal";
+import { Box, Button, Typography } from "@mui/material/";
+import { errorModalsAreSuppressed, type ErrorRecord, ErrorState, toggleSuppressErrorModals } from "./ErrorState";
+import { Router } from "../ui/GameRoot";
+import { SimplePage, ToastVariant } from "@enums";
+import { useRerender } from "../ui/React/hooks";
+import { OptionSwitch } from "../ui/React/OptionSwitch";
+import { LogBoxEvents } from "../ui/React/LogBoxManager";
+import { recentScripts } from "../Netscript/RecentScripts";
+import { SnackbarEvents } from "../ui/React/Snackbar";
+import { Settings } from "../Settings/Settings";
+
+export function ErrorModal(): React.ReactElement {
+  const { classes } = useStyles();
+  const rerender = useRerender();
+  const [error, setError] = useState<ErrorRecord | null>(ErrorState.ActiveError);
+
+  useEffect(() => {
+    const listener = (newError: ErrorRecord) => {
+      if (newError.force || (Router.page() !== SimplePage.RecentErrors && !errorModalsAreSuppressed())) {
+        setError(newError);
+        rerender();
+      }
+    };
+    return ErrorState.ErrorUpdate.subscribe(listener);
+  }, [rerender]);
+
+  const onClose = (force = false): void => {
+    ErrorState.ActiveError && (ErrorState.ActiveError.unread = false);
+
+    if (force || errorModalsAreSuppressed()) {
+      ErrorState.ActiveError = null;
+      setError(null);
+    } else {
+      const nextError = ErrorState.Errors.find((e) => e.unread) ?? null;
+      ErrorState.ActiveError = nextError;
+      setError(nextError);
+    }
```

Comment:
```suggestion
    }
    ErrorState.UnreadErrors = ErrorState.Errors.filter((e) => e.unread).length;
```
Bug (maybe): Run the test script. When you close each error, the number of errors in the sidebar is not changed.

IMO, this is a bug. That number is an indication of how many errors that the player has *not* been informed about. If they are looking at the error popup, read the bug description, then close the error popup, it means they are informed about the problems.